app/configurator/interactive-filters/time-slider.tsx

```diff
@@ -12,57 +12,29 @@ import { useTimeFormatUnit } from "@/formatters";
 import { DimensionMetadataFragment, TimeUnit } from "@/graphql/query-hooks";
 import { TemporalDimension } from "@/graphql/resolver-types";
 import { Icon } from "@/icons";
+import { Timeline } from "@/utils/observables";
 import useEvent from "@/utils/use-event";
 
-// TODO: make this configurable
-const ANIMATION_DURATION = 10000;
-
 type SliderDatum = {
   ms: number;
   formattedDate: string;
 };
 
-type TimeSliderState = {
-  progress: number;
-  setProgress: React.Dispatch<React.SetStateAction<number>>;
-  animating: boolean;
-  setAnimating: React.Dispatch<React.SetStateAction<boolean>>;
-};
+const TimelineContext = React.createContext<Timeline | undefined>(undefined);
 
-const TimeSliderStateContext = React.createContext<TimeSliderState>({
-  progress: 0,
-  setProgress: () => {},
-  animating: false,
-  setAnimating: () => {},
-});
+const useTimeline = () => {
+  // This is a "hack" to force re-rendering of the component using the hook.
+  const [_, setS] = React.useState(0);
```

Comment:
I think on React 18 we could remove the need for the stack hack through useSyncExternalStore.

https://beta.reactjs.org/reference/react/useSyncExternalStore#usesyncexternalstore

```
class Timeline extends Observable {
  getSnapshot() {
   // hopefully returning the timeline itself is OK for a snapshot
    return this
  }
}

const useTimeline = () => {
  const timeline = React.useContext(TimelineContext);
  if (!timeline) {
    throw new Error("useTimeline must be called inside a TimelineContext.Provider")
  }
  useSyncExternalStore(timeline.subscribe, timeline.getSnapshot)
  ...
}
```

Reply:
It seems a package offers a polyfill for older versions of React, I think it's worth it trying it out.

Reply:
Cool, another nice suggestion :) I've modified the code to use this approach, thanks!